File: pymod/gcp.py
from google.cloud import storage

class GCP:
	'''
	GCP
	'''
	def download_file(self, bucket_name, storage_key, local_file):
		'''
		download_file
		'''
		client = storage.Client()
		bucket = client.get_bucket(bucket_name)
		blob = bucket.blob(storage_key)
		blob.download_to_filename(local_file)

	def upload_file(self, bucket_name, local_file, storage_key):
		'''
		upload_file
		'''
		client = storage.Client()
		bucket = client.get_bucket(bucket_name)
		blob = bucket.blob(storage_key)
		blob.upload_from_filename(local_file)


'''
    def list_buckets(self):
	return  0
'''

File: src/main/java/skyapp/GCP.java
package skyapp;

import java.nio.file.Files;
import java.nio.file.Path;

import com.google.cloud.storage.BlobInfo;
import com.google.cloud.storage.Storage;
import com.google.cloud.storage.StorageOptions;

public class GCP {
    public void uploadFile(String bucketName, Path localFile, String storageKey) {
        byte[] content = null;
        try {
            content = Files.readAllBytes(localFile);
        } catch (Exception ex) {
            throw new Error(ex.getMessage(), ex);
        } finally {
            BlobInfo bi = BlobInfo.newBuilder(bucketName, storageKey).build();
            Storage storage = StorageOptions.getDefaultInstance().getService();
            storage.create(bi, content);
        }
    }

    public void downloadFile(String bucketName, String storageKey, Path localFile) {
        Storage storage = StorageOptions.getDefaultInstance().getService();
        storage.create(BlobInfo.newBuilder(bucketName, storageKey).build()).downloadTo(localFile);
    }
}